lib/samson/hooks.rb

```diff
@@ -25,6 +25,53 @@ def active?
       def require
         super @path
       end
+
+      def add_lib_path
+        engine.config.autoload_paths += Dir["#{engine.config.root}/lib/**/"]
+      end
+
+      def add_decorators
+        engine.config.after_initialize do
+          Dir.glob(engine.config.root.join('app/decorators/**/*_decorator.rb')).each do |c|
+            require_dependency(c)
+          end
+        end
+      end
+
+      def precompile_assets
+        assets = javascripts + stylesheets
+        engine.config.assets.precompile << assets
+      end
+
+      def javascripts
+        assets(:javascripts, :js)
+      end
+
+      def stylesheets
+        assets(:stylesheets, :scss, :css)
+      end
+
+      private
+
+      def assets(type, *exts)
+        basedir = assets_root(type)
+        return [] if basedir.nil?
+        find_assets(basedir, exts).map { |script| script.relative_path_from(Pathname.new(basedir)) }.map(&:to_s)
+      end
+
+      def find_assets(basedir, exts)
+        exts.map do |ext|
+          Dir.glob("#{basedir}/**/*.#{ext}").map { |asset| Pathname.new(asset) }
+        end.flatten
+      end
+
+      def assets_root(type)
```

Comment:
can this just do `engine.paths['app/assets/javascripts']` and return if it does not exist ? (avoid include logic / checking all paths for existence)